src/matteruser.coffee

```diff
@@ -89,6 +91,39 @@ class Matteruser extends Adapter
 
     send: (envelope, strings...) ->
         @client.postMessage(str, envelope.room) for str in strings
+        # Check if the target room is also a user's username
+        user = @robot.brain.userForName(envelope.room)
+
+        # If it's not, continue as normal
+        unless user
+            @client.postMessage(str, envelope.room) for str in strings
+            return
+
+        # If it is, we assume they want to DM that user
+        # Message their DM channel ID if it already exists.
+        if user.mm?.dm_channel_id?
+            @client.postMessage(str, user.mm.dm_channel_id) for str in strings
+            return
+
+        # Otherwise, create a new DM channel ID and message it.
+        @client.getUserDirectMessageChannel user.id, (data) =>
```

Comment:
Would be nice to move the error handling logic directly into `mattermost-client` as every client will need to implement the error 500 case.